### PR TITLE
[#82] Update the project configurations

### LINKED ISSUE #82: Update the project configurations
## Why

Currently, the project configurations are not good enough. We need to improve:
- Add `buildSrc` as dependency management for the project.
- Remove lint steps for deploying workflows.
- Add lint and detekt for kotlin source code.

## Who Benefits?

Developers

### PR DESCRIPTION
**Pull request by suho**
Resolves #82 

## What happened 👀

- Added buildSrc as dependency management for the project.
- Removed lint steps for deploying workflows.
- Added lint and detekt for kotlin source code.

## Insight 📝

`N/A`

## Proof Of Work 📹

Check the CI workflow for more detail

### CODE REVIEW

**Review by edgarss — APPROVED** [review]

File: android/build.gradle.kts [hunk 1]
```diff
@@ -1,72 +1,64 @@
...
     defaultConfig {
         applicationId = "co.nimblehq.ic.kmm.suv.android"
-        minSdk = 23
-        targetSdk = 32
-        versionCode = 1
-        versionName = "0.1.0"
+        minSdk = Version.ANDROID_MIN_SDK_VERSION
+        targetSdk = Version.ANDROID_TARGET_SDK_VERSION
+        versionCode = Version.ANDROID_VERSION_CODE
+        versionName = Version.ANDROID_VERSION_NAME
```
Comment (luongvo) [inline android/build.gradle.kts:33]: Do you intend to fix this issue https://github.com/suho/kmm-ic/pull/80#discussion_r963301813 in this PR too? On both production and staging build workflows 🤔 

```
        with:
          gradlePath: android/build.gradle.kts
          versionCode: ${{ github.run_number }}
```

**Review by luongvo — COMMENTED** [review]
@suho only 1 issue here

File: android/build.gradle.kts (hunk 1 above)
Comment (suho) [reply android/build.gradle.kts:33]: Good catch! I updated it here: https://github.com/suho/kmm-ic/pull/83/commits/22945828d69ca242652197b0e5f03624fe60eefe

Not override the versionName :+1:

**Review by luongvo — APPROVED** [review]
LGTM now

**Review by nkhanh44 — APPROVED** [review]

**Pull request merged**